### DISCUSSION TITLE
Can't get UseModalNavigation to work

### DISCUSSION THREAD

**Discussion by dzebas** [PrismLibrary/Prism#3069] [.NET MAUI] [answered]
Hi,

We are currently evaluating converting our huge enterprise app from Xamarin.Forms to Maui. We will become paying Prism customer as soon as we confirm that we can convert our app.

We have a fundamental issue at the very beginning - in old app we could push modal pages without issues, but with Prism.Maui it does nothing, for example adding to navigation parameters doesn't do anything for both animation or modal:

```
        navigationParameters = new NavigationParameters()
        {
            {KnownNavigationParameters.Animated, animated},
            {KnownNavigationParameters.UseModalNavigation, modal}
        };
```
This is how we start application:

```
public static MauiApp CreateMauiApp()
    {
        var builder = MauiApp.CreateBuilder();

        builder
            .UseMauiApp<App>()
            .UsePrism(prism =>
            {
                PrismStartup.Configure(prism, "/LoginPage");
            })
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
            });

        return builder.Build();
    }

    public static void Configure(PrismAppBuilder builder, string startingPage)
    {
        builder.RegisterTypes(RegisterTypes)
            .OnAppStart(startingPage);
    }


```

We are currently using Prism 9.0.271-pre. Are we doing something wrong or modal navigation doesn't work in Prism 9?

**Comment by brianlagunas** [comment]
Don't use the KnownNavigationParameters. Use the string instead. This has been fixed in the latest builds that are available to our Commercial Plus license holders.

**Comment by dzebas** [comment]
We really appreciate your prompt response. We were considering a commercial license, but at this point we are concerned that Prism has fundamental issues which suggest that product might not be production ready. We have other fundamental issues as well which we will need to resolve before we commit to Prism.

Is it possible to get trial license so we can evaluate if Prism is suitable for our needs?

**Reply by brianlagunas** [reply]
What fundamental issues are you concerned about specifically? If you want access to the nightly builds, private discord, tooling, and early releases you must purchase a Commercial Plus license.

**Reply by dzebas** [reply]
We are not interested in nightly builds, just a stable long term support release.

The other significant issue is that GoBackAsync is not working for us (only tested on iOS) - same code works fine with Prism 8 in Xamarin.Forms.

From your reply we assume that working trial version is not possible, so we have to consider all other options.

**Reply by brianlagunas** [reply]
Those issues do not exist in the builds available to license holders. The community has to wait longer to get stable builds. The trial is the community packages. If you want bugs fixes and patches sooner you would need a commercial plus license.

**Reply by dzebas** [reply]
Thank you Brian. Do we get source code with commercial plus license? In Prism 8 we had to patch PageUtilities.cs to work with DevExpress TabPage, so we need source code.

**Reply by brianlagunas** [reply, marked as answer]
Yes, the source code is available.

**Reply by MartinRobins** [reply]
Is the source here in github the same source as is used for the builds available to license holders or is this updated independently and then ported back here when the community builds are released?
